apps/shooting-clubs/acceptance/features/location-address.js

```diff
@@ -134,11 +134,66 @@ Scenario('from manual entry step: I am taken to the location-add-another-address
   I.seeInCurrentUrl(locationAddressPage.addAnotherAddress.url);
 });
 
-Scenario('add another address step: the correct form elements are present', (
+
+Scenario('category step: the correct form elements are present', (
+  I,
+  locationAddressPage
+) => {
+  I.seeElements(
+    [locationAddressPage.category.field.fullBoreRifles],
+    [locationAddressPage.category.field.smallBoreRifles],
+    [locationAddressPage.category.field.muzzleLoadingPistols]
+  )
+});
+
+Scenario('category step: I see an error when I submit an empty form', (
+  I,
+  locationAddressPage
+) => {
+  locationAddressPage.selectAddressAndSubmit();
+  I.submitForm();
+  I.seeErrors(locationAddressPage.category.field.locationAddressCategory)
+});
+
+Scenario('category step: when I select full-bore rifles category, I see this category against the address', (
+  I,
+  locationAddressPage
+) => {
+  locationAddressPage.selectAddressCategoryAndSubmit();
+  I.see(locationAddressPage.category.content.fullBoreRifles, locationAddressPage.addAnotherAddress.field.summary)
+});
+
+Scenario('category step: when I select 2 categories: full-bore rifles, small-bore rifles category, I see these categories against the address', (
+  I,
+  locationAddressPage
+) => {
+  locationAddressPage.selectAddressAndSubmit();
+  I.click(locationAddressPage.category.field.fullBoreRifles);
+  I.click(locationAddressPage.category.field.smallBoreRifles);
+  I.submitForm();
+  I.see(locationAddressPage.category.content.fullBoreRifles, locationAddressPage.addAnotherAddress.field.summary)
+  I.see(locationAddressPage.category.content.smallBoreRifles, locationAddressPage.addAnotherAddress.field.summary)
```

Comment:
I could use .seeEach with an array but this does not take a selector.  It just searches the whole page for text which can cause failures of tests if texts are duplicated.  It is also a major reason for slowing down acceptance tests for the registered traveller project in the past.